src/ParallelAlgorithms/Amr/Projectors/Tensors.hpp

```diff
@@ -0,0 +1,74 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <unordered_map>
+#include <utility>
+
+#include "DataStructures/ApplyMatrices.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Projection.hpp"
+#include "ParallelAlgorithms/Amr/Protocols/Projector.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace amr::projectors {
+
+/// \brief Update the Tensors corresponding to TensorTags after an AMR
+/// change
```

Comment:
Could you add a bit more detail? At least mention that the tensors are projected between meshes. Also mention `amr::projectors::ProjectVariables`. Also document the template params (TensorTags can be a typelist or pack, etc).